plugins/elanfp/fu-elanfp-device.c

```diff
@@ -40,40 +55,72 @@ fu_elanfp_iap_send_command(FuElanfpDevice *self,
 			   guint8 request,
 			   const guint8 *buf,
 			   gsize bufsz,
+			   gsize rspsz,
 			   GError **error)
 {
 	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(self));
 	gsize actual = 0;
-	guint8 buftmp[61] = {request, 0};
+	gsize sendsz = bufsz + 1;
+	guint8 start_index = 0x01;
+	guint8 buftmp[64] = {request, 0};
+	gboolean is_bulk_transfer = FALSE;
+
+	if (fu_device_has_private_flag(FU_DEVICE(self), FU_ELAN_FP_DEVICE_FLAG_USB_BULK_TRANSFER)) {
```

Comment:
I think just use fu_device_has_private_flag() directly, there's no need for a is_bulk_transfer temp variable. It's really just a bitfield check of a uint64 under the covers, and it's really fast.

Reply:
Fixed in 7ee92a89640014327469a87cc9be9a48998b484c.